### PR TITLE
Add NMEA Signal Quality

### PR DESCRIPTION
**Pull request by tracernz**
Add NMEA $GPGSV packet decoding to give satellite PRN numbers and SNR.
Tidy up the NMEA code a little but it could do with a lot more in the
future..
Tested with my MTK GPS which outputs data as per the NMEA standard (http://www.trimble.com/OEM_ReceiverHelp/V4.44/en/NMEA-0183messages_GSV.html)

### CODE REVIEW

File: src/main/io/gps.c [hunk 1]
```diff
@@ -514,130 +517,177 @@ static uint32_t grab_fields(char *src, uint8_t mult)
...
-                }
-                break;
-            case 7:
-                gps_Msg.numSat = grab_fields(string, 0);
-                break;
-            case 9:
-                gps_Msg.altitude = grab_fields(string, 0);     // altitude in meters added by Mis
-                break;
-            }
+        case '$':
```
Comment (ledvinap) [inline src/main/io/gps.c]: Maybe you can keep the style where case is on same indentation level as switch. It is used quite consistently in CF ...

Comment (ledvinap) [inline src/main/io/gps.c]: Sorry, checked now indented style is more common ... so this change is probably OK (and only diff is hard to read)

**Comment by tracernz** [comment]
You're right, it is hard to read unfortunately. New code is line 455, 537-8, 555-6, 603-645. :smile:

**Comment by tracernz** [comment]
Okay, I've split this into two separate commits so it's easier to read. :smile:

**Comment by hydra** [comment]
Merged, thanks for splitting up the whitespace changes into a new commit.  Made it much easier to review.

**Pull request merged**